Browser/keywords/browser_control.py

```diff
@@ -125,84 +151,147 @@ def take_screenshot(
+        mask = pos_params.get("mask", mask)
+        omitBackground = pos_params.get("omitBackground", omitBackground)
+        file_name = (
+            uuid.uuid4().hex
+            if filename is None or self._is_embed(filename)
+            else filename
+        )
         string_path_no_extension = str(
-            self._get_screenshot_path(filename, fileType.name)
+            self._get_screenshot_path(file_name, fileType.name)
         )
         with self.playwright.grpc_channel() as stub:
-            options = {
-                "path": f"{string_path_no_extension}.{fileType.name}",
-                "fileType": fileType.name,
-                "fullPage": fullPage,
-                "timeout": int(self.get_timeout(timeout)),
-                "omitBackground": omitBackground,
-            }
-            if mask:
-                mask_selectors: Optional[List[str]]
-                if isinstance(mask, str):
-                    mask_selectors = [self.resolve_selector(mask)]
-                elif isinstance(mask, Iterable):
-                    mask_selectors = [self.resolve_selector(str(s)) for s in mask]
-                else:
-                    raise ValueError(
-                        f"'mask' argument is neither string nor list of string. It is {type(mask)}"
-                    )
-            else:
-                mask_selectors = None
-
-            if quality is not None:
-                options["quality"] = max(min(100, quality), 0)
-            if disableAnimations:
-                options["animations"] = "disabled"
-            if crop:
-                options["clip"] = crop
+            options = self._create_screenshot_options(
+                crop,
+                disableAnimations,
+                fileType,
+                fullPage,
+                omitBackground,
+                quality,
+                string_path_no_extension,
+                timeout,
+            )
             response = stub.TakeScreenshot(
                 Request().ScreenshotOptions(
                     selector=self.resolve_selector(selector) or "",
-                    mask=json.dumps(mask_selectors),
+                    mask=json.dumps(self._get_mask_selectors(mask)),
                     options=json.dumps(options),
                     strict=self.strict_mode,
                 )
             )
             logger.debug(response.log)
-            file_path = response.body
-
-            if self._is_embed(filename):
-                return self._embed_to_log(file_path)
-            return self._log_image_link(file_path)
-
-    def _log_image_link(self, file_path: str) -> str:
+            screenshot_path_str = response.body
+            screenshot_path = Path(screenshot_path_str)
+            if (log_screenshot and self._is_embed(filename)) or return_as in (
+                ScreenshotReturnType.bytes,
+                ScreenshotReturnType.base64,
+            ):
+                with screenshot_path.open("rb") as png_file:
```

Comment:
read_bytes might be handy too: https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_bytes